lib/spack/spack/spec.py

```diff
@@ -1620,10 +1663,10 @@ def add_dependency_edge(self, dependency_spec: "Spec", *, deptypes: dp.Dependenc
                 # If we are here, it means the edge object was previously added to
                 # both the parent and the child. When we update this object they'll
                 # both see the deptype modification.
-                edge.add_type(deptypes)
+                edge.update_deptypes(deptypes=deptypes)
```

Comment:
I think we should be able to update the virtual information here as well, is there a reason that's not implemented as part of this?

Reply:
Done in 427bb95c56ce9aa8ad0af54ba25e40108795f860